packages/firebase_ui_firestore/lib/src/query_builder.dart

```diff
@@ -544,3 +709,38 @@ class _AggregateQueryBuilderState extends State<AggregateQueryBuilder> {
     }
   }
 }
+
+/// It takes in the child you want to show and calls the
+/// onBuild function when the child's build method is completed. This means
+/// when the child's is created and build method is called for the first time,
+/// we'll get this callback.
+/// Borrowed the idea from the link below and built on it further:
+/// https://www.filledstacks.com/post/how-to-perform-real-time-pagination-with-firestore/?utm_source=pocket_reader
+class BuildAwareWidget extends StatefulWidget {
```

Comment:
This naming seems confusing to me. Without reading the code, I'd assume `onBuild` is called after every build, not after widget mount. So maybe something like `AfterMountListener` will be a better name.

Reply:
How about `OnMountListener`? `onBuild` will also be renamed as `onMount`.